pkg/envoy/registry/registry.go

```diff
@@ -9,16 +9,21 @@ import (
 	"github.com/openservicemesh/osm/pkg/envoy"
 )
 
+// NewProxyRegistry initializes a new empty *ProxyRegistry.
+func NewProxyRegistry() *ProxyRegistry {
+	return &ProxyRegistry{}
+}
+
 // ExpectProxy catalogs the fact that a certificate was issued for an Envoy proxy and this is expected to connect to XDS.
-func (mc *MeshCatalog) ExpectProxy(cn certificate.CommonName) {
-	mc.expectedProxies.Store(cn, expectedProxy{
+func (pr *ProxyRegistry) ExpectProxy(cn certificate.CommonName) {
```

Comment:
ExpectProxy is not called anywhere (I think?). This was mostly used when injector was around OSM in the same binary as injector could fill the list of expected proxies to connect.
IMO we should remove this one, a proxy connecting or not it's the proxy's business. There are cleanup issues if a proxy is deleted before it connects (code as is)

Reply:
The only instance I see is here in a comment, pending #2481: https://github.com/openservicemesh/osm/blob/06ff02253408f7e322453b280c7dddd4108889e3/pkg/debugger/proxy.go#L37-L38

Do you still think that's safe to remove?

Reply:
We added ExpectProxy with the intent to provide debuggability.  We know that we issued a cert for a given proxy (via OSM Injector), now we can display that via the debug endpoint and show that we know of it but never connected.  It is just one more data point.  I don't think this is crucial for the functioning of the mesh, but it is a very nice to have for debuggability.

@nojnhuh I'd keep it, unless you think it actually removes value (makes the codebase more complex than it has to be)

Reply:
@nojnhuh As I finished typing this and submitted - I realized why this is no longer in use - after we moved Envoy sidecar injector into a separate pod, this made it harder to just call `ExpectProxy` (now it is a different pod).

I'd be supportive to remove this since it is dead code at this point and it will require a bit of work to get it to work as intended.

Reply:
Sounds good. I've removed it.